tests/integration_tests/build/test_coverage.py

```diff
@@ -80,20 +85,61 @@ def test_coverage(monkeypatch, record_property, metrics):
     '
     )
 
+    # Ignore code not relevant for the intended platform
+    # - CPUID and CPU template
+    # - Static CPU templates intended for specific CPU vendors
     # Generate coverage report.
-    utils.run_cmd(
-        f"""
-        grcov . \
-            -s . \
-            --binary-path ./build/cargo_target/{TARGET}/debug/ \
-            --excl-start "mod tests" \
-            --ignore "build/*" \
-            --ignore "**/tests/*" \
-            --ignore "**/test_utils*" \
-            -t html \
-            --ignore-not-existing \
-            -o ./build/cargo_target/{TARGET}/debug/coverage"""
-    )
+    if "AMD" == VENDOR:
+        utils.run_cmd(
+            f"""
+            grcov . \
+                -s . \
+                --binary-path ./build/cargo_target/{TARGET}/debug/ \
+                --excl-start "mod tests" \
+                --ignore "build/*" \
+                --ignore "**/tests/*" \
+                --ignore "**/test_utils*" \
+                --ignore "***/intel*" \
+                --ignore "****/intel*" \
+                --ignore "*t2*" \
+                --ignore "*t2s*" \
+                --ignore "*t2cl*" \
+                --ignore "*c3*" \
+                -t html \
+                --ignore-not-existing \
+                -o ./build/cargo_target/{TARGET}/debug/coverage"""
+        )
+    if "Intel" == VENDOR:
+        utils.run_cmd(
+            f"""
+            grcov . \
+                -s . \
+                --binary-path ./build/cargo_target/{TARGET}/debug/ \
+                --excl-start "mod tests" \
+                --ignore "build/*" \
+                --ignore "**/tests/*" \
+                --ignore "**/test_utils*" \
+                --ignore "***/amd*" \
+                --ignore "****/amd*" \
```

Comment:
I've got some reservations about not being quite precise about code paths that we are ignoring. But this is very easy to change later.

Reply:
Is there something special about using three or four `*` in a glob pattern that makes it different from just two?

Reply:
Some documentation I saw indicated it would only apply the string pattern at a certain depth of directories as I wanted this to only be applied to sub-modules, but I don't think that is true (I have not tested this though). I have already changed this though,filtering out by CPU vendor should be fairly comprehensive. A sub-folder called amd* on an intel host should not be applicable.